src/services/transaction_watcher_service.ts

```diff
@@ -0,0 +1,116 @@
+    private readonly _provider: SupportedProvider;
+    private readonly _web3Wrapper: Web3Wrapper;
+
+    constructor(dbConnection: Connection, provider: SupportedProvider) {
+        this._transactionRepository = dbConnection.getRepository(TransactionEntity);
+        this._provider = provider;
+        this._web3Wrapper = new Web3Wrapper(this._provider);
+    }
+    public async startAsync(): Promise<void> {
+        while (true) {
+            // finding transactions that need updating for which we have missed
+            // notifications from blocknative.
+            logger.trace('syncing transaciton status');
+            try {
+                await this.syncTransactionStatusAsync();
+            } catch (err) {
+                logger.error(`order watcher failed to sync transaction status`, { err: err.stack });
+            }
+            await utils.delay(TX_WATCHER_POLLING_INTERVAL_IN_MS);
+        }
+    }
+    public async syncTransactionStatusAsync(): Promise<void> {
+        await this._syncTransactionStatusAsync();
+    }
+
+    public async _syncTransactionStatusAsync(): Promise<void> {
+        const transactionsToCheck = await this._transactionRepository.find({
+            where: [
+                { status: TransactionStates.Unsubmitted },
+                { status: TransactionStates.Submitted },
+                { status: TransactionStates.Mempool },
+                { status: TransactionStates.Stuck },
+            ],
+        });
+        logger.trace(`found ${transactionsToCheck.length} transactions to check status`);
+        for (const tx of transactionsToCheck) {
+            await this._findTransactionStatusAndUpdateAsync(tx);
+        }
+    }
+
+    private async _findTransactionStatusAndUpdateAsync(tx: TransactionEntity): Promise<TransactionEntity> {
+        // TODO(oskar) - we are checking expiry inline, because TypeORM failed
+        // to work with LessThanOrEqual:
+        // const now = new Date();
+        // const transactionsToCheck = await this._transactionRepository.find({
+        //     where: [
+        //         {
+        //             status: TransactionStates.Mempool,
+        //             expiresAt: LessThanOrEqual(now),
+        //         },
+        //         {
+        //             status: TransactionStates.Submitted,
+        //             expiresAt: LessThanOrEqual(now),
+        //         },
+        //     ],
+        // });
+        const now = new Date();
+        const isExpired = tx.expectedAt <= now;
+        try {
+            const txInBlockchain = await this._web3Wrapper.getTransactionByHashAsync(tx.hash);
+            if (txInBlockchain !== undefined && txInBlockchain !== null && txInBlockchain.hash !== undefined) {
+                if (txInBlockchain.blockNumber !== null) {
+                    logger.info(
+                        `a transaction with a ${tx.status} status is already on the blockchain, updating status to confirmed`,
```

Comment:
This seems outdated now that this isn't just a cleanup job, but run as a regular cron. This case is expected now that we aren't using BlockNative, right? If so, let's remove this log.

Reply:
I've moved this log to a `logger.trace` as there might be a benefit to including this information when testing locally.